pallets/automation-time/src/tests.rs

```diff
@@ -253,19 +250,167 @@ fn run_tasks_completes_some_tasks() {
+			]
+		);
+		assert_eq!(1, AutomationTime::get_overflow_tasks().unwrap().len());
+	})
+}
+
+#[test]
+fn trigger_tasks_completes_all_overflow() {
+	new_test_ext().execute_with(|| {
+		let scheduled_time = SCHEDULED_TIME + 540;
+		let start_block_time: u64 = (scheduled_time) * 1000;
+
+		let message_one: Vec<u8> = vec![2, 4, 5];
+		create_overflow_task(ALICE, scheduled_time, message_one.clone());
+		let message_two: Vec<u8> = vec![2, 4];
+		create_overflow_task(ALICE, scheduled_time, message_two.clone());
+		Timestamp::set_timestamp(start_block_time);
+		System::reset_events();
+
+		AutomationTime::trigger_tasks(80_000);
+
+		assert_eq!(
+			events(),
+			[
+				Event::AutomationTime(crate::Event::Notify { message: message_one.clone() }),
+				Event::AutomationTime(crate::Event::Notify { message: message_two.clone() }),
+			]
+		);
+		assert_eq!(0, AutomationTime::get_overflow_tasks().unwrap().len());
+	})
+}
+
+#[test]
+fn trigger_tasks_does_some_of_both() {
+	new_test_ext().execute_with(|| {
+		let scheduled_time = SCHEDULED_TIME + 540;
+		let start_block_time: u64 = (scheduled_time) * 1000;
+
+		let message_one: Vec<u8> = vec![2, 4, 5];
+		create_overflow_task(ALICE, scheduled_time, message_one.clone());
+		let message_two: Vec<u8> = vec![2, 4];
+		schedule_task(ALICE, scheduled_time, message_two.clone());
+		let message_three: Vec<u8> = vec![2];
+		schedule_task(ALICE, scheduled_time, message_three.clone());
+		Timestamp::set_timestamp(start_block_time);
+		System::reset_events();
+
+		AutomationTime::trigger_tasks(80_000);
+
+		assert_eq!(
+			events(),
+			[
+				Event::AutomationTime(crate::Event::Notify { message: message_one.clone() }),
+				Event::AutomationTime(crate::Event::Notify { message: message_two.clone() }),
+			]
+		);
+		assert_eq!(0, AutomationTime::get_overflow_tasks().unwrap().len());
+	})
+}
+
+#[test]
+fn trigger_tasks_pushes_all_to_overflow() {
+	new_test_ext().execute_with(|| {
+		let scheduled_time = SCHEDULED_TIME + 540;
+		let start_block_time: u64 = (scheduled_time + 52) * 1000;
+
+		let message_one: Vec<u8> = vec![2, 4, 5];
+		create_overflow_task(ALICE, scheduled_time, message_one.clone());
+		let message_two: Vec<u8> = vec![2, 4];
+		create_overflow_task(ALICE, scheduled_time, message_two.clone());
+		let message_three: Vec<u8> = vec![2];
+		let expected_task_id = schedule_task(ALICE, scheduled_time, message_three.clone());
+		Timestamp::set_timestamp(start_block_time);
+		System::reset_events();
+
+		AutomationTime::trigger_tasks(80_000);
+
+		assert_eq!(
+			events(),
+			[
+				Event::AutomationTime(crate::Event::Notify { message: message_one.clone() }),
+				Event::AutomationTime(crate::Event::Notify { message: message_two.clone() }),
+			]
+		);
+		assert_eq!(expected_task_id, AutomationTime::get_overflow_tasks().unwrap()[0]);
+		assert_eq!(1, AutomationTime::get_overflow_tasks().unwrap().len());
+	})
+}
+
+#[test]
+fn trigger_tasks_pushes_some_to_overflow() {
```

Comment:
probably unlikely, what about 2 overflow consecutive? i know we execute the overflow tasks first, but what happens if we have too much volume when there's constant overflow?

Reply:
overflows can also be weight based right?

Reply:
the only way to protect against that is by adding buffers to our max scheduled tasks. An example would be allowing only 80% of the max tasks to be scheduled for a given slot. Since we cannot know a backup will occur ahead of time, we need to be very careful when we define our limits and safety buffers